wpilibNewCommands/src/test/java/edu/wpi/first/wpilibj2/command/SchedulingRecursionTest.java

```diff
@@ -115,6 +129,88 @@ public void end(boolean interrupted) {
     }
   }
 
+  @Test
+  void cancelFromEndLoop() {
+    try (CommandScheduler scheduler = new CommandScheduler()) {
+      AtomicInteger counter = new AtomicInteger();
+      Command dCancelsAll =
+          new Command() {
+            @Override
+            public void end(boolean interrupted) {
+              counter.incrementAndGet();
+              scheduler.cancelAll();
+            }
```

Comment:
All of these can be made with existing factories/classes rather than anonymous classes.